pkg/client/unversioned/horizontalpodautoscaler.go

```diff
@@ -40,12 +40,14 @@ type HorizontalPodAutoscalerInterface interface {
 
 // horizontalPodAutoscalers implements HorizontalPodAutoscalersNamespacer interface
 type horizontalPodAutoscalers struct {
-	client *ExtensionsClient
+	// Use RESTClient instead of more specific to allow reusing the type in different clients.
+	// TODO(piosz): change to AutoscalingClient once completely migrated to 'autoscaling' group.
+	client *RESTClient
```

Comment:
what is this needed for? I'd rather take two clients in the short term than expose RESTClient here
cc @deads2k